src/models.py

```diff
@@ -496,52 +493,60 @@ def build_reddit_module(task, d_inp, params):
 
 
 def build_image_sent_module(task, d_inp, params):
-    pooler = Pooler.from_params(d_inp, params['d_proj'])
+    pooler = Pooler(d_inp, d_proj=params['d_proj'])
     return pooler
 
 
-def build_single_sentence_module(task, d_inp, params):
+def build_single_sentence_module(task, d_inp, use_bert, params):
     ''' Build a single classifier '''
-    pooler = Pooler.from_params(d_inp, params['d_proj'])
-    classifier = Classifier.from_params(params['d_proj'], task.n_classes, params)
+    pool_type = "first" if use_bert else "max"
+    pooler = Pooler(d_inp, project=not use_bert, d_proj=params['d_proj'], pool_type=pool_type)
+    d_out = d_inp if use_bert else params["d_proj"]
+    classifier = Classifier.from_params(d_out, task.n_classes, params)
     return SingleClassifier(pooler, classifier)
 
 
-def build_pair_sentence_module(task, d_inp, model, vocab, params):
+def build_pair_sentence_module(task, d_inp, vocab, params):
     ''' Build a pair classifier, shared if necessary '''
 
     def build_pair_attn(d_in, use_attn, d_hid_attn):
-        ''' Build the pair model '''
-        if not use_attn:
-            pair_attn = None
-        else:
-            d_inp_model = 2 * d_in
-            modeling_layer = s2s_e.by_name('lstm').from_params(
-                Params({'input_size': d_inp_model, 'hidden_size': d_hid_attn,
-                        'num_layers': 1, 'bidirectional': True}))
-            pair_attn = AttnPairEncoder(vocab, modeling_layer,
-                                        dropout=params["dropout"])
+        ''' Build the pair attn module '''
+        d_inp_model = 2 * d_in
+        modeling_layer = s2s_e.by_name('lstm').from_params(
+            Params({'input_size': d_inp_model, 'hidden_size': d_hid_attn,
+                    'num_layers': 1, 'bidirectional': True}))
+        pair_attn = AttnPairEncoder(vocab, modeling_layer, dropout=params["dropout"])
         return pair_attn
 
-    if params["attn"]:
-        pooler = Pooler.from_params(params["d_hid_attn"], params["d_hid_attn"], project=False)
+    # Build the "pooler", which does pools a variable length sequence
+    #   possibly with a projection layer beforehand
+    if params["attn"] and not model.use_bert:
```

Comment:
this is the relevant BERT change: to stick as close to their fine-tuning scheme as possible, we don't want to do a projection after pooling